app/src/main/java/uw/hcrlab/kubi/lesson/HintArrayAdapter.java

```diff
@@ -0,0 +1,41 @@
+package uw.hcrlab.kubi.lesson;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+
+import uw.hcrlab.kubi.R;
+
+/**
+ * Created by Alexander on 4/28/2016.
+ */
+public class HintArrayAdapter extends ArrayAdapter<PromptData.Hint> {
+    private final Context context;
+    private final ArrayList<PromptData.Hint> hints;
+
+    public HintArrayAdapter(Context context, ArrayList<PromptData.Hint> hints) {
```

Comment:
Because ArrayAdapters only take List<> or [] as a collection to operate over. We can pass in the HintCollection and then pass the hints array into the super constructor. I just didn't think of that. I'm not really sure why we have a HintCollection at all though instead of simply having an ArrayList<PromptData.Hint>. It doesn't seem like there is any point to HintCollection right now except to wrap an ArrayList<>. Did you have a reason for creating that type (I don't remember the original name before I renamed it to HintCollection)?